### PR TITLE
Add `ApplicationBuilder.(get_updates_)socket_options`

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
* Follow up for #3935
* Also adds tests that make sure that `ApplicationBuilder` has all necessary methods

- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [x] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- ~Added myself alphabetically to ``AUTHORS.rst`` (optional)~
-  ~Added new classes & modules to the docs and all suitable ``__all__`` s~
-  [x] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

### CODE REVIEW

**Review by lemontree210 — COMMENTED** [review]
I did look through the code and it looks fine but I don't feel like I'm qualified enough to judge :)

File: tests/ext/test_applicationbuilder.py [hunk 1]
```diff
@@ -379,6 +416,36 @@ class Client:
...
+            "request"
+        ).get_updates_socket_options(((4, 5, 6),)).get_updates_connection_pool_size(
+            "get_updates"
+        ).build()
+
+        for kwargs in httpx_request_kwargs:
+            if kwargs.get("connection_pool_size") == "request":
+                assert kwargs.get("socket_options") == ((1, 2, 3),)
+            else:
+                assert kwargs.get("socket_options") == ((4, 5, 6),)
```
Comment (harshil21) [inline tests/ext/test_applicationbuilder.py:447]:
```suggestion
                assert kwargs.get("get_updates_socket_options") == ((4, 5, 6),)
```
shouldn't it be this?

**Review by harshil21 — COMMENTED** [review]
code in general looks good to me.

File: tests/ext/test_applicationbuilder.py (hunk 1 above)
Comment (harshil21) [reply tests/ext/test_applicationbuilder.py:447]: oh nvm it is kwargs from HTTPXRequest, that parameter doesn't exist

**Pull request merged**